### PR TITLE
Refactor KubernetesClusterResourceModifierActionWorker

### PR DESCRIPTION
**Pull request by FelipeM525**
### Description

This PR aims to refactor the class `KubernetesClusterResourceModifierActionWorker`. The main goal is to improve the adherence to object-oriented programming principles, specifically encapsulation. In this class, there were a couple of methods that were using reflection, such as `Field.setAccessible()`, to modify object fields directly, which can be seen as a violation of encapsulation, one of the fundamental principles of object-oriented programming. This issue was fixed by creating setters in the respective classes belonging to the objects where this behavior occurred.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

### How Has This Been Tested?
I ran all the tests related to the class `KubernetesClusterResourceModifierActionWorker` and certified none of them were broken.

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,21 @@
+cloudstack (4.20.0.0-SNAPSHOT~focal~focal~focal) unstable; urgency=low
```
Comment (weizhouapache) [inline debian/changelog:1]: please remove these changes

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java [hunk 2]
```diff
@@ -111,8 +111,29 @@
...
 import com.cloud.vm.dao.VMInstanceDao;
+import org.apache.cloudstack.api.ApiConstants;
+import org.apache.cloudstack.api.BaseCmd;
+import org.apache.cloudstack.api.command.user.firewall.CreateFirewallRuleCmd;
+import org.apache.cloudstack.api.command.user.network.CreateNetworkACLCmd;
+import org.apache.cloudstack.api.command.user.vm.StartVMCmd;
+import org.apache.cloudstack.api.command.user.volume.ResizeVolumeCmd;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
```
Comment (DaanHoogland) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:122]: there are quite some unused and duplicate imports in this file, @FelipeM525 . Can you deal with those?

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java [hunk 3]
```diff
@@ -785,32 +768,32 @@ protected String getKubernetesClusterNodeNamePrefix() {
...
+            if (autoscaleEnabled != null) {
+                updatedCluster.setAutoscalingEnabled(autoscaleEnabled.booleanValue());
+            }
+            updatedCluster.setMinSize(minSize);
+            updatedCluster.setMaxSize(maxSize);
+            return kubernetesClusterDao.persist(updatedCluster);
         });
     }
 
+
```
Comment (GaOrtiga) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:795]: @FelipeM525 Can you remove these extra lines?

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Review by JoaoJandre — APPROVED** [review]
CLGTM, nice cleanup

**Comment by DaanHoogland** [comment]
@weizhouapache @GaOrtiga , is this good for your sakes?

**Comment by GaOrtiga** [comment]
> @weizhouapache @GaOrtiga , is this good for your sakes?

Yes, CLTM

**Review by GaOrtiga — APPROVED** [review]

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java [hunk 4]
```diff
@@ -551,60 +536,58 @@ protected void removePortForwardingRules(final IpAddress publicIp, final Network
         for (PortForwardingRuleVO pfRule : pfRules) {
             if (startPort <= pfRule.getSourcePortStart() && pfRule.getSourcePortStart() <= endPort) {
                 portForwardingRulesDao.remove(pfRule.getId());
+                logger.debug(String.format("The Port forwarding rule [%s] with the id [%s] was removed.", pfRule.getName(), pfRule.getId()));
```
Comment (BryanMLima) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:539]: No need to use `String.format()` here, with the updated log4j.

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java [hunk 5]
```diff
@@ -537,6 +521,7 @@ protected void removePortForwardingRules(final IpAddress publicIp, final Network
                 for (PortForwardingRuleVO pfRule : pfRules) {
                     if (pfRule.getVirtualMachineId() == vmId) {
                         portForwardingRulesDao.remove(pfRule.getId());
+                        logger.debug(String.format("The Port forwarding rule [%s] with the id [%s] was removed.", pfRule.getName(), pfRule.getId()));
```
Comment (BryanMLima) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:524]: No need to use `String.format()` here, with the updated log4j.

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java [hunk 6]
```diff
@@ -524,6 +507,7 @@ protected FirewallRule removeSshFirewallRule(final IpAddress publicIp) {
             if (firewallRule.getSourcePortStart() == CLUSTER_NODES_DEFAULT_START_SSH_PORT) {
                 rule = firewallRule;
                 firewallService.revokeIngressFwRule(firewallRule.getId(), true);
+                logger.debug(String.format("The SSH firewall rule [%s] with the id [%s] was revoked",firewallRule.getName(),firewallRule.getId()));
```
Comment (BryanMLima) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:510]: No need to use `String.format()` here, with the updated log4j.

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java [hunk 7]
```diff
@@ -511,6 +493,7 @@ protected FirewallRule removeApiFirewallRule(final IpAddress publicIp) {
                     firewallRule.getSourcePortEnd() == CLUSTER_API_PORT) {
                 rule = firewallRule;
                 firewallService.revokeIngressFwRule(firewallRule.getId(), true);
+                logger.debug(String.format("The API firewall rule [%s] with the id [%s] was revoked",firewallRule.getName(),firewallRule.getId()));
```
Comment (BryanMLima) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:496]: No need to use `String.format()` here, with the updated log4j.

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java [hunk 8]
```diff
@@ -551,60 +536,58 @@ protected void removePortForwardingRules(final IpAddress publicIp, final Network
...
-                    rule.getSourcePortEnd() == CLUSTER_API_PORT) {
-                lbService.deleteLoadBalancerRule(rule.getId(), true);
-                break;
-            }
-        }
+        List<LoadBalancerVO> loadBalancerRules = loadBalancerDao.listByIpAddress(publicIp.getId());
+        loadBalancerRules.stream().filter(lbRules -> lbRules.getNetworkId() == network.getId() && lbRules.getAccountId() == account.getId() && lbRules.getSourcePortStart() == CLUSTER_API_PORT
+        && lbRules.getSourcePortEnd() == CLUSTER_API_PORT).forEach(lbRule -> {
+            lbService.deleteLoadBalancerRule(lbRule.getId(), true);
+            logger.debug(String.format("The load balancing rule with the Id: %s was removed",lbRule.getId()));
```
Comment (BryanMLima) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:551]: No need to use `String.format()` here, with the updated log4j.

**Review by BryanMLima — APPROVED** [review]
Nice clean up, CLGTM, just a few remarks in the logs.

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java (hunk 8 above)
Comment (DaanHoogland) [reply plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:551]:
```suggestion
            logger.debug("The load balancing rule with the Id: {} was removed",lbRule.getId());
```

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java (hunk 4 above)
Comment (DaanHoogland) [reply plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:539]:
```suggestion
                logger.debug("The Port forwarding rule [{}] with the id [{}] was removed.", pfRule.getName(), pfRule.getId());
```

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java (hunk 5 above)
Comment (DaanHoogland) [reply plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:524]: see https://github.com/apache/cloudstack/pull/8801#discussion_r1639482249

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java (hunk 6 above)
Comment (DaanHoogland) [reply plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:510]: see https://github.com/apache/cloudstack/pull/8801#discussion_r1639482249

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java (hunk 7 above)
Comment (DaanHoogland) [reply plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java:496]: see https://github.com/apache/cloudstack/pull/8801#discussion_r1639482249

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@FelipeM525 you have unused imports, can you clean them, please?

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9

**Pull request merged**